### ISSUE TITLE
Support one time login emails

### ISSUE THREAD

**Issue by thobson** [passlock-dev/passkeys#6]
For devices that don't support passkeys we should offer single use login links/codes aka "magic emails".

There are some edge cases around this however:

1. Mallory registers a passkey against alice@gmail.com
2. Alice chooses to login via email code
3. We log her in (and mark the email as verified)
4. Mallory uses his passkey to log in to Alice's account 💥

### Preventing account takeover

**Passkey (unverified email) followed by email login**
Force the user to re-authenticate using a passkey. This is kind of counter intuitive because they're most likely trying to log in via email because they can't use a passkey for some reason. Unfortunately I can't think of a viable alternative.

**Passkey (verified email) followed by email login**
All good

**Email login (verified email) followed by passkey registration** 
Force the user to re-authenticate using an email code. The UX isn't great, however we may be able to mitigate this by storing a short lived token in local storage, similar to the way we handle mailbox verification.

### Social login

Basically the same approach. In most cases the Google email will be verified so it's scenario 2 above.
